sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -104,3 +110,37 @@ def test_md5sums():
         assert_false(sg.check_md5sum_change(f.name))
 
     os.remove(f.name + '.md5')
+
+
+def test_pattern_matching():
+    """Test if only examples matching pattern are executed"""
```

Comment:
> If you don't have docstrings, you don't see the test in the list when you run nosetests on the full file / folder.

In my experience if you don't have a docstring you see the fully qualified function name always. For example, here is the output of `nosetests sphinx_gallery/tests/test_gen_rst.py -v` which has some docstrings and some fully qualified function names:

```
Test if a known example file gets properly split ... ok
Test over the known requirements of supported syntax in the ... ok
test_gen_rst.test_direct_comment_after_docstring ... ok
Test the correct translation of a code block into rst ... ok
test_gen_rst.test_extract_intro ... ok
Test md5sum check functions work on know file content ... ok
```

I don't think it's a nosetests bug per se but just a nosetests choice. It may make sense for some but I find it annoying, hence I never use docstring in test functions and always comment.

Oh well, let's try not to get too distracted on this side issue.